x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -407,17 +391,60 @@ && hasRemoteIndices(request.getRoleDescriptors())) {
                 return;
             }
 
-            final Set<RoleDescriptor> userRolesWithoutDescription = removeUserRoleDescriptorDescriptions(userRoleDescriptors);
-            final Set<RoleDescriptor> filteredUserRoleDescriptors = maybeRemoveRemotePrivileges(
-                userRolesWithoutDescription,
+            Set<RoleDescriptor> filteredRoleDescriptors = filterRoleDescriptorsForMixedCluster(
+                userRoleDescriptors,
                 transportVersion,
                 request.getId()
             );
 
-            createApiKeyAndIndexIt(authentication, request, filteredUserRoleDescriptors, listener);
+            createApiKeyAndIndexIt(authentication, request, filteredRoleDescriptors, listener);
         }
     }
 
+    private Set<RoleDescriptor> filterRoleDescriptorsForMixedCluster(
+        final Set<RoleDescriptor> userRoleDescriptors,
+        final TransportVersion transportVersion,
+        final String... apiKeyIds
+    ) {
+        final Set<RoleDescriptor> userRolesWithoutDescription = removeUserRoleDescriptorDescriptions(userRoleDescriptors);
+        Set<RoleDescriptor> filteredUserRoleDescriptors = maybeRemoveRemotePrivileges(
+            userRolesWithoutDescription,
+            transportVersion,
+            apiKeyIds
+        );
+        return maybeRemoveGlobalPrivileges(filteredUserRoleDescriptors, transportVersion, apiKeyIds);
```

Comment:
I believe we should fail the request in a mixed cluster scenario instead of filtering. The reason for filtering `description` and `remote_indices` was because we changed our internal and predefined roles to include these fields. Since predefined roles are out of user's control, we had to strip it to maintain BWC. But since we did not change any of our internal roles to include global manage indices privileges, we can simply fail create/update API key requests and not worry about filtering because we don't allow user roles to be created in a mixed cluster either.

Reply:
I see. That makes sense. I'll change it to fail if a user tries to create an API key with a manage roles privilege in a mixed cluster where it's not supported.